packages/salesforcedx-vscode-core/src/orgPicker/orgList.ts

```diff
@@ -0,0 +1,125 @@
+import { Aliases, AuthInfo } from '@salesforce/core';
+import {
+  CancelResponse,
+  ContinueResponse,
+  ParametersGatherer
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import fs = require('fs');
+import { isNullOrUndefined, promisify } from 'util';
+import { StatusBarAlignment, StatusBarItem, window } from 'vscode';
+import { getDefaultUsernameOrAlias } from '../context/workspaceOrgType';
+import { nls } from '../messages';
+
+export interface FileInfo {
+  scratchAdminUsername?: string;
+  isDevHub?: boolean;
+  username: string;
+}
+const readFileAsync = promisify(fs.readFile);
+export class OrgList {
+  public authFileObjects: FileInfo[];
+
+  public async getAuthInfoObjects() {
+    const authFilesArray = await AuthInfo.listAllAuthFiles();
+
+    if (authFilesArray === null || authFilesArray.length === 0) {
+      return Promise.reject(new Error('No previous authorizations found.'));
+    }
+    const authInfoObjects = Promise.all(
+      authFilesArray.map(fileName => {
+        const filePath = path.join(os.homedir(), '.sfdx', fileName);
+        return readFileAsync(filePath, 'utf8')
+          .then(fileData => JSON.parse(fileData))
+          .catch(err => null);
+      })
+    );
+    return authInfoObjects;
+  }
+
+  public async filterAuthInfo(authInfoObjects: FileInfo[]) {
+    authInfoObjects = authInfoObjects.filter(fileData =>
+      isNullOrUndefined(fileData.scratchAdminUsername)
+    );
+    authInfoObjects = authInfoObjects.filter(fileData =>
+      isNullOrUndefined(fileData.isDevHub)
```

Comment:
This filters out dev hubs from appearing in the dropdown list for orgs, because we only display the `defaultusername` field in the status bar. I can see why including the dev hubs would be useful as well if a developer wants to create a scratch org in a different dev hub.